public/app/features/alerting/unified/rule-list/PaginatedGrafanaLoader.tsx

```diff
@@ -0,0 +1,89 @@
+import { groupBy } from 'lodash';
+import { useEffect, useMemo, useRef } from 'react';
+
+import { Icon, Stack, Text } from '@grafana/ui';
+import { GrafanaRulesSourceSymbol } from 'app/types/unified-alerting';
+import { GrafanaPromRuleGroupDTO } from 'app/types/unified-alerting-dto';
+
+import { GrafanaRuleLoader } from './GrafanaRuleLoader';
+import { DataSourceSection } from './components/DataSourceSection';
+import { LazyPagination } from './components/LazyPagination';
+import { ListGroup } from './components/ListGroup';
+import { ListSection } from './components/ListSection';
+import { RuleGroupActionsMenu } from './components/RuleGroupActionsMenu';
+import { useGrafanaGroupsGenerator } from './hooks/prometheusGroupsGenerator';
+import { usePaginatedPrometheusGroups } from './hooks/usePaginatedPrometheusGroups';
+
+export const GROUP_PAGE_SIZE = 40;
+
+export function PaginatedGrafanaLoader() {
+  const grafanaGroupsGenerator = useGrafanaGroupsGenerator();
+
+  const groupsGenerator = useRef(grafanaGroupsGenerator(GROUP_PAGE_SIZE));
+
+  useEffect(() => {
+    const currentGenerator = groupsGenerator.current;
+    return () => {
+      currentGenerator.return();
+    };
+  }, []);
+
+  const {
+    page: groupsPage,
+    nextPage,
+    previousPage,
+    canMoveForward,
+    canMoveBackward,
+    isLoading,
+  } = usePaginatedPrometheusGroups(groupsGenerator.current, GROUP_PAGE_SIZE);
+
+  const groupsByFolder = useMemo(() => groupBy(groupsPage, 'folderUid'), [groupsPage]);
+
+  return (
+    <DataSourceSection name="Grafana" application="grafana" uid={GrafanaRulesSourceSymbol} isLoading={isLoading}>
```

Comment:
nit: translation here on the name

Reply:
To clarify; translation for "Grafana"? 😄

Reply:
😂 Ha yes, fair point, I guess it doesn't need to be translated when the section name is literally just "Grafana"